File: server/image/resize.js
var execFileSync=require('child_process').execFileSync
var fs=require('fs')
var img=fs.readFileSync("1.jpg")
var stdout=execFileSync("./resize",["-w=50","-h=50","-t=jpg"],{input:img,maxBuffer: 20*1024*1024})
fs.writeFileSync("2.jpg",stdout)

File: server/image/resize.go
package main

import (
	"flag"
	"github.com/nfnt/resize"
	"image/jpeg"
	"image/png"
	"os"
)

func main() {
	w := flag.Uint("w", 0, "width")
	h := flag.Uint("h", 0, "height")
	ty := flag.String("t", "jpg", "image type")
	flag.Parse()
	if *ty == "png" {
		src, err := png.Decode(os.Stdin)
		if err != nil {
			panic(err)
		}
		dst := resize.Resize(*w, *h, src, resize.Lanczos3)
		png.Encode(os.Stdout, dst)
	} else if *ty == "jpg" {
		src, err := jpeg.Decode(os.Stdin)
		if err != nil {
			panic(err)
		}
		dst := resize.Resize(*w, *h, src, resize.Lanczos3)
		jpeg.Encode(os.Stdout, dst, nil)
	}

}
